### PR TITLE
UPDATE: configurable auth provider

### PR DESCRIPTION
**Pull request by vkislichenko**
I'm sure there is a valid reasoning behind current logic of extending **User** model with shopify shop trait,
But I really do need them to be separate.

This PR provides one more shopify app config option `shop_auth_provider` ("users" by default)
Model is then taken from config `auth.providers.{$config['shop_auth_provider']}.model`

Works if manual migrations are used and user table is replaced with custom one
Models provided by this package will still reference custom model using `user_id` columns, but it seems like a minor detail related to bigger refactoring.

### CODE REVIEW

**Comment by osiset** [comment]
Sounds good to me! Please see the failed lint, adjust, and I'll merge :)

**Comment by vkislichenko** [comment]
UPDATE: added config option `shop_auth_guard` 
It's mostly to prevent use of custom default auth guards (like jwt), but it might be also useful in other cases when `shop_auth_provider` specifies a custom model

File: src/Http/Middleware/AuthProxy.php [hunk 1]
```diff
@@ -82,6 +82,11 @@ public function handle(Request $request, Closure $next)
         // Login the shop
         $shop = $this->shopQuery->getByDomain($shop);
         if ($shop) {
+            // Override auth guard
+            if(($guard = Util::getShopifyConfig('shop_auth_guard'))) {
```
Comment (osiset) [inline src/Http/Middleware/AuthProxy.php:86]: @vkislichenko Fails linting here, needs a space after the if

File: src/Http/Middleware/VerifyShopify.php [hunk 2]
```diff
@@ -248,6 +248,11 @@ protected function loginShopFromToken(SessionToken $token, NullableSessionId $se
             return false;
         }
 
+        // Override auth guard
+        if(($guard = Util::getShopifyConfig('shop_auth_guard'))) {
```
Comment (osiset) [inline src/Http/Middleware/VerifyShopify.php:252]: Same as previous @vkislichenko , needs space after if to pass linting.

**Review by osiset — CHANGES_REQUESTED** [review]
Couple small issues to correct linting, then can be merged.

**Review by osiset — APPROVED** [review]

**Comment by osiset** [comment]
Checks passed, linting pass. Merging. Thanks @vkislichenko !

**Comment by salahy** [comment]
What about the migrations ? they still change the user model

**Comment by vkislichenko** [comment]
@salahy 
> What about the migrations ? they still change the user model

there's "Manual migrations" in config
and command `php artisan vendor:publish --tag=shopify-migrations` to push migrations to your app's folder so you're free to modify before migrating.

**Comment by nahid** [comment]
When this changes will be released? @osiset

**Pull request merged**